internal/runners/projects/edit.go

```diff
@@ -0,0 +1,179 @@
+package projects
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/prompt"
+	"github.com/ActiveState/cli/internal/rtutils/p"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_client/projects"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/ActiveState/cli/pkg/projectfile"
+)
+
+const (
+	visibilityPublic  = "Public"
+	visibilityPrivate = "Private"
+)
+
+type EditParams struct {
+	Namespace   *project.Namespaced
+	ProjectName string
+	Visibility  string
+	Repository  string
+}
+
+func (e EditParams) validate() error {
+	if e.Visibility != "" &&
+		!strings.EqualFold(e.Visibility, visibilityPublic) &&
+		!strings.EqualFold(e.Visibility, visibilityPrivate) {
+		return locale.NewInputError("err_edit_visibility", "Visibility must be either public or private")
+	}
+
+	return nil
+}
+
+type Edit struct {
+	auth   *authentication.Auth
+	out    output.Outputer
+	prompt prompt.Prompter
+	config configGetter
+}
+
+func NewEdit(prime primeable) *Edit {
+	return &Edit{
+		auth:   prime.Auth(),
+		out:    prime.Output(),
+		prompt: prime.Prompt(),
+		config: prime.Config(),
+	}
+}
+
+func (e *Edit) Run(params EditParams) error {
```

Comment:
Do we need an `e.auth.Authenticated()` check here and error out ASAP if it fails?